### PR TITLE
Feature/si 2087 definition minting manufacturer privilege

### PR DESCRIPTION
**Pull request by LorranSutter**

### CODE REVIEW

**Review by elffjs — APPROVED** [review]

File: contracts/implementations/tableland/DeviceDefinitionTable.sol [hunk 1]
```diff
@@ -128,30 +132,42 @@ contract DeviceDefinitionTable is AccessControlInternal {
...
+        uint256 manufacturerId,
         DeviceDefinitionInput calldata data
     ) external {
         ITablelandTables tablelandTables = TablelandDeployments.get();
-        string memory prefix = DeviceDefinitionTableStorage
-            .getStorage()
-            .prefixes[tableId];
+        DeviceDefinitionTableStorage.Storage
+            storage dds = DeviceDefinitionTableStorage.getStorage();
+        uint256 tableId = dds.tables[manufacturerId];
```
Comment (elffjs) [inline contracts/implementations/tableland/DeviceDefinitionTable.sol:149]: Could we bail here if this is zero? Can Tableland tables be burned?

File: contracts/implementations/tableland/DeviceDefinitionTable.sol [hunk 2]
```diff
@@ -128,30 +132,42 @@ contract DeviceDefinitionTable is AccessControlInternal {
...
+     * @param manufacturerId The unique identifier of the manufacturer
      * @param data Input data with the following fields:
      *  id -> The alphanumeric ID of the Device Definition
      *  model -> The model of the Device Definition
      *  year -> The year of the Device Definition
      *  metadata -> The metadata stringfied object of the Device Definition
      */
     function insertDeviceDefinition(
-        uint256 tableId,
+        uint256 manufacturerId,
```
Comment (elffjs) [inline contracts/implementations/tableland/DeviceDefinitionTable.sol:143]: I think this is a good change.

File: contracts/implementations/tableland/DeviceDefinitionTable.sol [hunk 3]
```diff
@@ -128,30 +132,42 @@ contract DeviceDefinitionTable is AccessControlInternal {
...
         try INFT(address(tablelandTables)).ownerOf(tableId) returns (
             address tableIdOwner
         ) {
-            if (msg.sender != tableIdOwner) {
+            INFTMultiPrivilege manufacturerIdProxy = INFTMultiPrivilege(
+                ManufacturerStorage.getStorage().idProxyAddress
+            );
+
+            if (
+                msg.sender != tableIdOwner &&
```
Comment (elffjs) [inline contracts/implementations/tableland/DeviceDefinitionTable.sol:160]: Could we get rid of this line? The table NFT ownership is sort of external to us. We're being asked to manage the table using DIMO access control.

File: contracts/implementations/tableland/DeviceDefinitionTable.sol (hunk 1 above)
Comment (LorranSutter) [reply contracts/implementations/tableland/DeviceDefinitionTable.sol:149]: The contracts does not allow tables burning. We would catch that in the `ownerOf` below anyway.

File: contracts/implementations/tableland/DeviceDefinitionTable.sol (hunk 3 above)
Comment (LorranSutter) [reply contracts/implementations/tableland/DeviceDefinitionTable.sol:160]: Yeah makes sense

**Pull request merged**